queenbee/schema/artifact_location.py

```diff
@@ -9,11 +9,28 @@
     3. S3: An S3 bucket
 """
 from queenbee.schema.qutil import BaseModel
+import queenbee.schema.variable as qbvar
 from pydantic import Field, constr
-from typing import Dict
+from typing import Dict, List
 from enum import Enum
 
 
+def referenced_values(values) -> Dict[str, List[str]]:
+    """Get referenced variables if any"""
+    ref_values = {}
+    
+    if not values:
+        return ref_values
+
+    for value in values:
+        if value is None:
+            continue
+        ref_var = qbvar.get_ref_variable(value)
+        if ref_var:
+            ref_values[value] = ref_var
+
+    return ref_values
```

Comment:
I think we should either make this function  hidden and use it only for internal use in this module or move it to a helper module.

Reply:
Maybe even add it as a static method to the BaseModel and reference it from these classes?